src/CMakeLists.txt

```diff
@@ -637,9 +667,14 @@ install(TARGETS ceph-mon DESTINATION bin)
 if(NOT ALLOCATOR STREQUAL "jemalloc")
   set(disable_jemalloc "DISABLE_JEMALLOC=1")
 endif()
+set(EXTRA_CXXFLAGS -fPIC -Wno-unused-variable -DNDEBUG)
```

Comment:
@tchaikov 
Sorry, I'm a bit thick in the head today.

Because there is Linux-ism code used when NDEBUG is off. 
But I'll check with the newly import rocksdb sha.
On the other hand if NDEBUG is defined for the project, is is silly to actually compile rocksdb with NDEBUG.
I'll move a beter version of this to #10418 as well